accounts-db/src/accounts.rs

```diff
@@ -552,31 +595,54 @@ impl Accounts {
         account_locks: &mut AccountLocks,
         writable_keys: Vec<&Pubkey>,
         readonly_keys: Vec<&Pubkey>,
-    ) -> Result<()> {
+        batch_account_locks: &mut BatchAccountLocks,
+        allow_self_conflicting_entries: bool,
+    ) -> (Result<()>, bool) {
+        let mut self_conflicting_tx = false;
```

Comment:
nit: this indiciates a self-conflicting batch. tx still cannot have duplicate keys so can't conflict with itself.